### ISSUE TITLE
NXmx: should NXdetector_channel in NXdetector be required?

### ISSUE THREAD

**Issue by spc93** [nexusformat/definitions#1511]
This field probably doesn't make sense in many types of detector so would it be better to make it optional to avoid adding meaningless values or giving a validation error?

**Comment by phyy-nx** [comment]
Telco: this looks like a bug.  Will check with dectris then add recommended=True.

**Comment by soph-dec** [comment]
Agreed, it makes no sense that this field is required, this is a bug. I think making it optional is enough though.
